### PR TITLE
Release

### PR DESCRIPTION
**Pull request by omkar273**
## Summary by CodeRabbit

* **New Features**
  * Support for mixed billing cadences within a single subscription (e.g., monthly + quarterly line items)
  * Invoice previews and windowed period previews for mixed-period billing

* **Changes**
  * Validation enforces alignment of billing intervals and blocks proration when mixed cadences are present
  * New billing-period and cancellation semantics (additional cancel-at options)

* **Documentation**
  * Added comprehensive PRD describing multi-cadence billing, edge cases, and migration guidance

* **Tests**
  * Added unit/integration/e2e tests covering multi-cadence scenarios and proration rules

### CODE REVIEW

File: internal/service/subscription_change_test.go [hunk 1]
```diff
@@ -1228,6 +1286,8 @@ func (s *SubscriptionChangeServiceTestSuite) TestFixedToUsagePlanTransition() {
 // 	}
 // }
 
+// TestSubscriptionChangeServiceTestSuite runs the subscription change suite.
+// Uncomment to run; requires integration deps (e.g. IntegrationFactory) for full subscription creation.
 // func TestSubscriptionChangeServiceTestSuite(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (omkar273) [reply internal/service/subscription_change_test.go:1291]: This was commented previously

**Review by nkmishra1997 — APPROVED** [review]
